test/utility.cpp

```diff
@@ -91,6 +91,304 @@ TEST_CASE("exchange", "[utility.exchange]") {
+//   constexpr remove_reference_t<T>&& move(T&&) noexcept;
+TEST_CASE("move", "[forward]") {
+  struct S {};
+
+  // lvalue -> rvalue
+  static S s;
+  static S const cs{};
+
+  /* type */ {
+    CHECK(std::is_same<decltype(slb::move(s)), S&&>::value);
+    CHECK(std::is_same<decltype(slb::move(cs)), S const&&>::value);
+  }
+  /* `noexcept` */ {
+    CHECK(noexcept(slb::move(s)));
+    CHECK(noexcept(slb::move(cs)));
+  }
+#ifndef _MSC_VER
+  /* C++11 `constexpr` */ {
+    constexpr S&& a = slb::move(s);
+    constexpr S const&& b = slb::move(cs);
+    (void)a;
+    (void)b;
+  }
+#endif
+
+  // rvalue -> rvalue
+  struct {
+    constexpr S operator()() const noexcept { return S{}; }
+  } make_s;
+
+  struct {
+    constexpr S const operator()() const noexcept { return S{}; }
+  } make_cs;
+
+  /* type */ {
+    CHECK(std::is_same<decltype(slb::move(make_s())), S&&>::value);
+    CHECK(std::is_same<decltype(slb::move(make_cs())), S const&&>::value);
+  }
+  /* `noexcept` */ {
+    CHECK(noexcept(slb::move(make_s())));
+    CHECK(noexcept(slb::move(make_cs())));
+  }
+  /* C++11 `constexpr` */ {
+#if __has_warning("-Wpessimizing-move")
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpessimizing-move"
+#endif
+    constexpr S a = slb::move(make_s());
+    constexpr S b = slb::move(make_cs());
+#if __has_warning("-Wpessimizing-move")
+#pragma clang diagnostic pop
+#endif
+    (void)a;
+    (void)b;
+  }
+
+  CHECK_CXX14_CONSTEXPR(move, {
+    int x = 0;
+    int const cx = 1;
+    struct {
+      constexpr int operator()() const noexcept { return 2; }
+    } make_x{};
+    return slb::move(x) == 0 && slb::move(cx) == 1 && slb::move(make_x()) == 2;
+  });
+}
+
+template <bool IsNothrow = true>
+struct MoveNothrow {
+  MoveNothrow() = default;
+  MoveNothrow(MoveNothrow const&) {}
+  MoveNothrow(MoveNothrow&&) noexcept(IsNothrow) {}
+};
+
+// template<class T>
+//   constexpr conditional_t<
+//       !is_nothrow_move_constructible_v<T> && is_copy_constructible_v<T>,
+//       const T&,
+//       T&&>
+//     move_if_noexcept(T& x) noexcept;
+TEST_CASE("move_if_noexcept", "[forward]") {
+  struct MoveOnly {
+    MoveOnly() = default;
+    MoveOnly(MoveOnly const&) = delete;
+    MoveOnly(MoveOnly&&) {}
+  };
+
+  struct Legacy {
+    Legacy() = default;
+    Legacy(Legacy const&) {}
+  };
+
+  // lvalue -> rvalue
+  static MoveNothrow<true> mnt;
+  static MoveOnly mo;
+  static MoveOnly const cmo{};
+
+  /* type */ {
+    CHECK(std::is_same<decltype(slb::move_if_noexcept(mnt)),
+                       MoveNothrow<true>&&>::value);
+    CHECK(std::is_same<decltype(slb::move_if_noexcept(mo)), MoveOnly&&>::value);
+    CHECK(std::is_same<decltype(slb::move_if_noexcept(cmo)),
+                       MoveOnly const&&>::value);
+  }
+  /* `noexcept` */ {
+    CHECK(noexcept(slb::move_if_noexcept(mnt)));
+    CHECK(noexcept(slb::move_if_noexcept(mo)));
+    CHECK(noexcept(slb::move_if_noexcept(cmo)));
+  }
+#ifndef _MSC_VER
```

Comment:
What's the motivation for skipping the check on msvc?

Reply:
Left a comment.